Parse/src/main/java/com/parse/ParseInstallation.java

```diff
@@ -143,9 +143,7 @@ public String getInstallationId() {
       @Override
       public Task<Void> then(Task<Void> task) throws Exception {
         // Retry the fetch as a save operation because this Installation was deleted on the server.
-        // Do not attempt to resave an object if LDS is enabled, since changing objectId is not allowed.
```

Comment:
So is it fairly easy to null out your objectId() on a failure and then cause multiple install id's per app?